backend/molgenis-emx2-graphql/src/main/java/org/molgenis/emx2/graphql/GraphqlTableFieldFactory.java

```diff
@@ -605,7 +605,12 @@ private DataFetcher fetcherForTableQueryField(Table aTable) {
         q.offset((int) args.get(GraphqlConstants.OFFSET));
       }
       if (args.containsKey(GraphqlConstants.ORDERBY)) {
-        q.orderBy((Map<String, Order>) args.get(GraphqlConstants.ORDERBY));
+        Map<String, Order> orderBy = (Map<String, Order>) args.get(ORDERBY);
+        Map<String, Order> unescapedMap = new HashMap<>();
+        for (var entry : orderBy.entrySet()) {
+          unescapedMap.put(entry.getKey().replace("__", "_"), entry.getValue());
```

Comment:
Hmm, I am a bit worried. 

I expected we would have an 'unescape' method to mirror the 'escape' method, with suitable tests
https://github.com/molgenis/molgenis-emx2/blob/4880b49b7496870209860b68ad3a9c767637645e/backend/molgenis-emx2-graphql/src/main/java/org/molgenis/emx2/graphql/GraphqlTableFieldFactory.java#L44

So minimally you need
- "replaceAll" and 
-  replace "_" with " ".

However, apparently we have solved this elsewhere in a different way. We used  ```findColumnById(aTable, s.getName());``` to check whether the column actually exists, and then to find its name. Because I remember the 'unescape' might not be deterministic (in theory) because '__' might come from a '_' or two spaces '  '.

See most ugly code in whole of emx2 here findColumnById is used for unescaping the query
https://github.com/molgenis/molgenis-emx2/blob/4880b49b7496870209860b68ad3a9c767637645e/backend/molgenis-emx2-graphql/src/main/java/org/molgenis/emx2/graphql/GraphqlTableFieldFactory.java#L525

For filters there is analogously ugly code here
https://github.com/molgenis/molgenis-emx2/blob/4880b49b7496870209860b68ad3a9c767637645e/backend/molgenis-emx2-graphql/src/main/java/org/molgenis/emx2/graphql/GraphqlTableFieldFactory.java#L440

So I suppose we need something like that for orderBy which I hope may be much simpler, and therefore most shorter? Or we might want to exand the 'filter' code to also apply to sort parameter?

Reply:
i think the above is incorrect , i have added a test and moved the unescape to a function ( not really needed imo as it really context dependant ). The unescape is a lot simpler than the escape. Please add a failing test case if you think some thing is missing